pages/api/users/index.js

```diff
@@ -6,23 +6,28 @@ import Profile from "../../../models/Profile";
 
 export default async function handler(req, res) {
   await connectMongo();
+
   const directoryPath = path.join(process.cwd(), "data");
-  const files = fs
-    .readdirSync(directoryPath)
-    .filter((item) => item.includes("json"));
+  const files = await fs.promises.readdir(directoryPath);
+
+  const users = await Promise.all(
+    files
+      .filter((item) => item.includes("json"))
+      .map(async (file) => {
+        const filePath = path.join(directoryPath, file);
+        try {
+          const data = await fs.promises.readFile(filePath, "utf8");
```

Comment:
I think this will block the `map`?

Reply:
Then what changes should I make in here?

Reply:
The async reading of the files should happen async, then loaded into the promise all